.github/workflows/build-publish.yaml

```diff
@@ -0,0 +1,65 @@
+name: Build - publish
+
+on:
+  push:
+    branches:
+      - main
+  workflow_dispatch:
+    inputs:
+      docker_tag_suffix:
+        description: "Enter a Docker Tag suffix (e.g dev)\nSuffixed images will not be auto-deployed"
+        required: false
+        type: string
+
+env:
+  ECR_REPOSITORY_URL: 673156464838.dkr.ecr.us-west-2.amazonaws.com
+  ECR_REPOSITORY_NAME: pgt-leader-bot
+
+jobs:
+  build-publish:
+    name: Build and Publish Docker Image
+    runs-on: minafoundation-default-runners
+    steps:
+      - name: 📥 Checkout
+        uses: actions/checkout@v4
+
+      - name: 🦀 Get application version from setup.py
+        id: application-version
+        run: |
+          version=$(grep "version" setup.py | head -1 | cut -d'"' -f2)
+          echo "current-version=$version" >> "$GITHUB_OUTPUT"
```

Comment:
@kaozenn I couldn't find another repo where we get the application version in a `setup.py`, so I improvised with this two lines.